cashu/wallet/helpers.py

```diff
@@ -0,0 +1,453 @@
+            )
+            if not is_api:
+                print(f"Received {sum_proofs(redeem_proofs)} sats")
+
+
+async def print_mint_balances(wallet, show_mints=False):
+    """
+    Helper function that prints the balances for each mint URL that we have tokens from.
+    """
+    # get balances per mint
+    mint_balances = await wallet.balance_per_minturl()
+
+    # if we have a balance on a non-default mint, we show its URL
+    keysets = [k for k, v in wallet.balance_per_keyset().items()]
+    for k in keysets:
+        ks = await get_keyset(id=str(k), db=wallet.db)
+        if ks and ks.mint_url != wallet.url:
+            show_mints = True
+
+    # or we have a balance on more than one mint
+    # show balances per mint
+    if len(mint_balances) > 1 or show_mints:
+        print(f"You have balances in {len(mint_balances)} mints:")
+        print("")
+        for i, (k, v) in enumerate(mint_balances.items()):
+            print(
+                f"Mint {i+1}: Balance: {v['available']} sat (pending: {v['balance']-v['available']} sat) URL: {k}"
+            )
+        print("")
+
+
+async def get_mint_wallet(wallet: Wallet, is_api: bool = False):
+    """
+    Helper function that asks the user for an input to select which mint they want to load.
+    Useful for selecting the mint that the user wants to send tokens from.
+    """
+    await wallet.load_mint()
+
+    mint_balances = await wallet.balance_per_minturl()
+
+    if len(mint_balances) > 1:
+        if not is_api:
+            await print_mint_balances(wallet, show_mints=True)
+
+        url_max = max(mint_balances, key=lambda v: mint_balances[v]["available"])
+        nr_max = list(mint_balances).index(url_max) + 1
+
+        if is_api:
+            mint_nr_str = None
+        else:
+            mint_nr_str = input(
+                f"Select mint [1-{len(mint_balances)}] or "
+                f"press enter for mint with largest balance (Mint {nr_max}): "
+            )
+        if not mint_nr_str:  # largest balance
+            mint_url = url_max
+        elif mint_nr_str.isdigit() and int(mint_nr_str) <= len(
+            mint_balances
+        ):  # specific mint
+            mint_url = list(mint_balances.keys())[int(mint_nr_str) - 1]
+        else:
+            if is_api:
+                raise HTTPException(
+                    status_code=status.HTTP_400_BAD_REQUEST, detail="invalid input."
+                )
+            else:
+                raise Exception("invalid input.")
+    else:
+        mint_url = list(mint_balances.keys())[0]
+    print(mint_url)
+    print(wallet.name)
+    # load this mint_url into a wallet
+    mint_wallet = Wallet(mint_url, os.path.join(settings.cashu_dir, wallet.name))
+    mint_keysets: WalletKeyset = await get_keyset(mint_url=mint_url, db=mint_wallet.db)  # type: ignore
+    print(mint_keysets)
+    print(mint_wallet)
+    # load the keys
+    assert mint_keysets.id
+    await mint_wallet.load_mint(keyset_id=mint_keysets.id)
```

Comment:
Needs to be rebased to #162 because of a bug (and some forgotten comments seem to be left).